src/Analysis/Ast/Impl/Indexing/IIndexManager.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using Microsoft.Python.Analysis.Documents;
+
+namespace Microsoft.Python.Analysis.Indexing {
+    internal interface IIndexManager {
+        void AddRootDirectory();
+        void ProcessFile(Uri uri, IDocument doc);
+        void ProcessClosedFile(Uri uri);
```

Comment:
Where are we going to use this method?

Reply:
Open, update and close events on files will signal the index manager. 
So, for example, if the file is not on the workspace, all  of the file's symbols will be removed